### PR TITLE
chore: cleanup whenCurrentJobsFinished

### PR DESCRIPTION
**Pull request by gabegorelick**
Cleanup and document whenCurrentJobsFinished. And since it's now official API, add some tests for it.

### CODE REVIEW

File: lib/queue.js [hunk 1]
```diff
@@ -1174,49 +1174,20 @@ Queue.prototype.clean = function(grace, type, limit) {
  * @returns {Promise}
  */
 Queue.prototype.whenCurrentJobsFinished = function() {
-  return new Promise((resolve, reject) => {
```
Comment (gabegorelick) [inline lib/queue.js]: Removing the unnecessary `new Promise`.

File: lib/queue.js [hunk 2]
```diff
@@ -1174,49 +1174,20 @@ Queue.prototype.clean = function(grace, type, limit) {
...
-        return forcedReconnection;
-      })
-      .then(resolve, reject);
-
-    /*
-    this.bclient.disconnect();
-    this.bclient.once('end', function(){
-      console.error('ENDED!');
-      setTimeout(function(){
-        this.bclient.connect();
```
Comment (gabegorelick) [inline lib/queue.js]: Removed all this commented code.

File: lib/queue.js [hunk 3]
```diff
@@ -1174,49 +1174,20 @@ Queue.prototype.clean = function(grace, type, limit) {
...
-      });
-      stream.on('error', function(err){
-        console.error('errir', err);
-        this.bclient.connect();
-      });
-      this.bclient.connect();
-    }
-    */
-    //this.bclient.connect();
+  return Promise.all([this.processing[0]]).then(() => {
```
Comment (holm) [inline lib/queue.js]: After upgrade to latest version of bull we are seeing that on shutdown, it doesn't properly wait for all jobs to complete. 

It seems to me that this is likely the change that caused the problem. Was it intentional to change this to only wait for the first job?

Comment (gabegorelick) [reply lib/queue.js]: No, that seems like a dumb mistake on my part. Thanks for catching it @holm.

Comment (gabegorelick) [reply lib/queue.js]: Fix: https://github.com/OptimalBits/bull/pull/1586

Comment (holm) [reply lib/queue.js]: Thanks for the quick fix!

**Pull request merged**